Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -722,6 +722,116 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Resolve Conflict SG", "[.SyncServe
+        {{ // three sets of braces? because Xcode
+            collectionSpecs[0], kC4Disabled, kC4OneShot,
+            nullslice, nullptr, _pullFilter, this
+        }}
+    };
+    C4ParamsSetter paramsSetter = [&replCollections](C4ReplicatorParameters& c4Params) {
+        c4Params.collectionCount = replCollections.size();
+        c4Params.collections = replCollections.data();
+    };
+
+    // Setup onDocsEnded:
+    _enableDocProgressNotifications = true;
+    _onDocsEnded = [](C4Replicator* repl,
+                      bool pushing,
+                      size_t numDocs,
+                      const C4DocumentEnded* docs[],
+                      void* context) {
+        for (size_t i = 0; i < numDocs; ++i) {
+            auto doc = docs[i];
+            if ((doc->flags & kRevPurged) == kRevPurged) {
+                ((ReplicatorAPITest*)context)->_docsEnded++;
+            }
+        }
+    };
+
+    auto collRoses = collections[0];
+
+    // Put doc in remote DB, in channels a and b
+    sendRemoteRequest("PUT", docIDstr, &status, &error, R"({"channels":["a", "b"],"scopes":{"flowers":{"collections":{"roses":{}}}}})");
+    REQUIRE(status == HTTPStatus::Created);
+
+    // Pull doc into CBL:
+    C4Log("-------- Pulling");
+    replicate(paramsSetter);
+
+    // Verify:
+    c4::ref<C4Document> doc1 = c4coll_getDoc(collRoses, slice(docIDstr), true, kDocGetAll, nullptr);
+    REQUIRE(doc1);
+    CHECK(slice(doc1->revID).hasPrefix("1-"_sl));
+    CHECK(_docsEnded == 0);
+    CHECK(_counter == 0);
+
+    // Revoked access to channel 'a':
+    sendRemoteRequest("PUT", "_user/purgeRevoke", &status, &error, R"({"admin_channels":["b"]})", true);
+    REQUIRE(status == HTTPStatus::OK);
+
+    // Check if update to doc1 is still pullable:
+    auto oRevID = slice(doc1->revID).asString();
+    sendRemoteRequest("PUT", docIDstr, R"({"_rev":")" + oRevID + R"(", "channels":["b"],"scopes":{"flowers":{"collections":{"roses":{}}}}})");
+
+    C4Log("-------- Pull update");
+    replicate(paramsSetter);
+
+    // Verify the update:
+    doc1 = c4coll_getDoc(collRoses, slice(docIDstr), true, kDocGetAll, nullptr);
+    REQUIRE(doc1);
+    CHECK(slice(doc1->revID).hasPrefix("2-"_sl));
+    CHECK(_docsEnded == 0);
+    CHECK(_counter == 0);
+
+    auto curRevID = slice(doc1->revID).asString();
+
+    // Revoke access to all channels:
+    sendRemoteRequest("PUT", "_user/purgeRevoke", &status, &error, R"({"admin_channels":[]})", true);
+    REQUIRE(status == HTTPStatus::OK);
+
+    C4Log("-------- Pull the revoked");
+    replicate(paramsSetter);
+
+    // Verify that doc1 is purged:
+    doc1 = c4coll_getDoc(collRoses, slice(docIDstr), true, kDocGetAll, nullptr);
+    REQUIRE(!doc1);
+    CHECK(_docsEnded == 1);
+    CHECK(_counter == 1);
+
+    // Purge remote doc so test will succeed multiple times without flushing bucket
+    sendRemoteRequest("POST", "_purge", &status, &error, "{\"" + docIDstr + "\":[\"*\"]}", true);
```

Comment:
In this case, you don't rely on the uniqueness of the docIDs, 
   const string docIDstr = timePrefix() + "apera-doc1"; 
right?

Reply:
Yeah, const string docIDstr = timePrefix() + "apera-doc1", so doc has unique ID.